src/tls_wrap.cc

```diff
@@ -323,6 +322,33 @@ void TLSWrap::EncOut() {
                                                                  &count);
   CHECK(write_size_ != 0 && count != 0);
 
+  // New Session Ticket
+  // https://www.ietf.org/rfc/rfc5077.txt
+  // TODO base this on the actual packet, not length
+  if (write_size_ == 258) {
```

Comment:
This is the real hack. We listen to all socket output for New Session Ticket events, and strip the IV from the ticket and call our function in JS land that simulates a newSession.